src/main/java/fi/okm/jod/yksilo/controller/ehdotus/MahdollisuudetController.java

```diff
@@ -120,28 +135,62 @@ public List<EhdotusDto> createEhdotus(@RequestBody @Valid LuoEhdotusDto ehdotus)
                 ehdotus.kiinnostusPainotus,
                 kiinnostukset.stream().map(o -> o.nimi().get(Kieli.FI)).toList()));
 
-    var result =
-        inferenceService.infer(endpoint, request, Response.class).stream()
-            .collect(Collectors.toMap(Suggestion::id, r -> r, (exising, newValue) -> exising));
+    return inferenceService.infer(endpoint, request, Response.class).stream()
+        .collect(Collectors.toMap(Suggestion::id, r -> r, (exising, newValue) -> exising));
+  }
+
+  private List<EhdotusDto> populateEmptyEhdotusDtos(
+      LinkedHashSet<UUID> ids, Set<UUID> tyomahdollisuusIds) {
 
+    // initialize lexicalIndex counters
+    final var tmLexicalIndex = new AtomicInteger();
+    final var kmLexicalIndex = new AtomicInteger();
     return ids.stream()
         .map(
-            key ->
-                Optional.ofNullable(result.get(key))
-                    .orElseGet(
-                        () -> {
-                          log.warn(
-                              "kohtaanto id {} not found from tyomahdollisuus or koulutus IDs",
-                              key.toString());
-                          var tyyppi = getTyyppi(key, tyomahdollisuusIds);
-                          return new Suggestion(key, -1, tyyppi.name());
-                        }))
+            key -> {
+              var type = getTyyppi(key, tyomahdollisuusIds);
+              int index;
+              switch (type) {
+                case TYOMAHDOLLISUUS -> index = tmLexicalIndex.getAndIncrement();
+                case KOULUTUSMAHDOLLISUUS -> index = kmLexicalIndex.getAndIncrement();
+                default -> throw new IllegalStateException("Unexpected value: " + type);
+              }
```

Comment:
This seems to produce separate interleaved ordering for työ- and koulutusmahdollisuus.That is, given `t1,t2,...,k1,k2,...`, the index-based ordering will be `t1(index=0),k1(index=0),t2(1),k2(1),...` where (in lexical order) `t1 <= t2` and `k1 <= k2`, but t1 might be > k1.

Reply:
This comparison will been taken care at the frontend, when t and k arrays are combined and visible set of one page opportunities is sorted by the actual string.

Reply:
This has been changed to combined lexical index of all opportunities.